compiler/transforms/set_memory_space.py

```diff
@@ -67,19 +67,19 @@ def change_to_memory_space(t):
 class InitMemRefGlobalMemorySpace(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: memref.GetGlobal, rewriter: PatternRewriter):
-        # global variables should go in memory space 0 (L3)
+        # global variables should go in memory space L3
         memspace = op.memref.type.memory_space
 
         # If memory space is already 0, don't do anything
-        if isinstance(memspace, builtin.IntegerAttr) and memspace.value.data == 0:
+        if isinstance(memspace, builtin.StringAttr) and memspace.data == "L3":
```

Comment:
This pattern repeats itself quite a few times. Maybe you can refactor this into a helper function?
```python
def is_in_snax_memory_space(memref_type, memspace: str):
     return isinstance(memref_type.memspace, builtin.StringAttr) and memref_type.memspace.data == str
```